app/jobs/get_suggestions_job.rb

```diff
@@ -0,0 +1,93 @@
+class GetSuggestionsJob < ApplicationJob
+  queue_as :default
+
+  def perform(job_status)
+    @job_id = job_status.job_id
+
+    send_spotify_request
+
+    if get_spotify_tracks
+      job_status.update(job_attributes_success)
+    else
+      job_status.update(job_attributes_failure)
+    end
+    job_status.done = true
+    job_status.save!
+  end
+
+  private
+
+  def client_token
+    Base64.strict_encode64(
+      "#{ENV['SPOTIFY_CLIENT_ID']}:#{ENV['SPOTIFY_CLIENT_SECRET']}",
```

Comment:
Might be good to use `Hash#fetch` so that you can't mistakenly not set these.

Power move: set a constant in an intiializer and use the constant from here.

`config/initializers/spotify.rb`:

```ruby
SPOTIFY_CLIENT_ID = ENV.fetch("SPOTIFY_CLIENT_ID")
SPOTIFY_CLIENT_SECRET = ENV.fetch("SPOTIFY_CLIENT_SECRET")
```

```ruby
def client_token
  Base64.strict_encode64([SPOTIFY_CLIENT_ID, SPOTIFY_CLIENT_SECRET].join(":")).delete("\n")
end
```

That way you discover configuration problems on app boot.